stellar-contract-env-common/src/env_val.rs

```diff
@@ -131,15 +133,15 @@ impl<E: Env> From<EnvVal<E, RawVal>> for RawVal {
 }
 
 impl<E: Env, T: TagType> TryFrom<EnvVal<E, RawVal>> for TaggedVal<T> {
-    type Error = ();
+    type Error = ConversionError;
```

Comment:
Is it practical to have a unique error type per type so that it is extra clear what conversion is failing? I think it might be worth making it different for non-wasm builds too and include an indication between what type the value actually is to speed up debugging.

Reply:
Possibly. I think we'll do another systematic pass over errors in the future before we finalize things to pick the right granularity -- the work here (along with the PR @jayz22 landed) is just to get us to a world where we're using typed errors split up into roughly the right shape.

Reply:
Created https://github.com/stellar/rs-stellar-contract-env/issues/158